applications/tari_merge_mining_proxy/src/error.rs

```diff
@@ -20,27 +20,58 @@
 // WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
 // USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-use tari_core::{chain_storage::ChainStorageError, proof_of_work::monero_rx::MergeMineError};
+use hex::FromHexError;
+use std::io;
+use tari_common::{ConfigError, ConfigurationError};
+use tari_core::{proof_of_work::monero_rx::MergeMineError, transactions::CoinbaseBuildError};
 use thiserror::Error;
+use tonic::transport;
 
 #[derive(Debug, Error)]
 pub enum MmProxyError {
-    #[error("Chain storage error: {source}")]
-    ChainStorageError {
-        #[from]
-        source: ChainStorageError,
-    },
+    #[error("Configuration error: {0}")]
+    ConfigurationError(#[from] ConfigurationError),
+    #[error("Configuration error: {0}")]
+    ConfigError(#[from] ConfigError),
```

Comment:
Weird that these two similarly named errors exist. I've created #2228 to merge them